### DISCUSSION TITLE
paddle模型转onnx模型报错：[ERROR] Cannot found attribute beta in op: swish

### DISCUSSION THREAD

**Discussion by pcycccccc** [PaddlePaddle/PaddleOCR#13232] [Q&A] [answered]
Version:
paddlepaddle-gpu-2.6.1
paddleocr 2.7.3
paddle2onnx 1.0.6

由于我需要将paddle文字识别模型转为onnx文件，进行c++推理，转换指令：
`paddle2onnx --model_dir inference_model_myself\rec --model_filename inference.pdmodel --params_filename inference.pdiparams --save_file liyi_rec.onnx`

但是报错：[ERROR] Cannot found attribute beta in op: swish

我测试了paddleIOCR官方提供的推理模型，将推理模型用该指令转为onnx模型，转换成功，并且可以正常推理。但是使用了我自己训练完的模型，却报这个错误。不是很懂为什么，查找了一下，将paddlepaddle退回2.5.2也无法解决这个问题

![d46c329b99073602f7606378fbaaadf](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/7413d5f2-d047-4ecc-aaa9-f5314b99dba8)

有没有大佬能给点建议！或者是否有其他转onnx模型的方法可以尝试的

**Comment by GreatV** [comment]
更新一下paddle2onnx版本

**Reply by pcycccccc** [reply]
我尝试了一下 通过pip install paddle2onnx更新了版本，好像已经是最新版本的了paddle2onnx 1.0.6，但还是有问题

**Reply by GreatV** [reply]
paddle版本也需要使用最新版本，这个问题应该之前paddle2onnx已经修复了。

**Reply by jzhang533** [reply]
paddle2onnx 最新的版本是 1.2.4 ，因为 pip 上现在没有 windows 的预编译包。需要你自行编译，或者换成用 linux 系统。

**Reply by pcycccccc** [reply]
paddlepaddle-gpu-2.6.1 和 paddlepaddle-gpu-3.0都试过了
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/a43015ca-adc8-4202-bc8b-8c16f5f3d727)
这个是用项目中的推理代码进行测试的，推理结果是正确的。
不知道，您那边能否帮我测试一下模型转换。
[rec.zip](https://github.com/user-attachments/files/16050317/rec.zip)

**Reply by pcycccccc** [reply]
@jzhang533  换成linux系统是否可以使用pip命令来安装最新版本的paddle2onnx呢？

**Reply by jzhang533** [reply, marked as answer]
> @jzhang533 换成linux系统是否可以使用pip命令来安装最新版本的paddle2onnx呢？

可以。 https://pypi.org/project/paddle2onnx/#files 这里只有 linux 下的预编译安装包。

**Reply by pcycccccc** [reply]
@jzhang533  好的 谢谢 我试一下！

**Reply by pcycccccc** [reply]
感谢！我已成功使用linux完成onnx模型的获取。另外，我想请教一下，如何导出onnx能够使得推理效果与直接使用paddle模型文件保持差不多的效果，我注意到说明上面提到了要动态转换，请问如何进行动态转换呢？
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/108455796/66832e7f-5630-4026-87da-425087823e45)

**Reply by SWHL** [reply]
这个转换工作，我之前有整理一个转换工具专门来做这个事情的，支持动态shape推理。
link: [PaddleOCRModelConvert](https://github.com/RapidAI/PaddleOCRModelConvert)

同时，基于onnx的推理代码也有整理好，参见[RapidOCR](https://github.com/RapidAI/RapidOCR)

**Reply by pcycccccc** [reply]
@SWHL  谢谢大佬！！
